pkg/datapath/loader/sock.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package loader
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/maps/lbmap"
+	"github.com/cilium/cilium/pkg/maps/socktermfilter"
+
+	"github.com/cilium/ebpf"
+)
+
+var (
+	BPFSockTermPath = "/bpf/bpf_sock_term.o"
+)
+
+type sockTermObjects struct {
+	SockUDPDestroy *ebpf.Program `ebpf:"cil_sock_udp_destroy"`
+	SockTermFilter *ebpf.Map     `ebpf:"cilium_sock_term_filter"`
+}
+
+// LoadSockTerm configures and loads the cil_sock_udp_destroy program and
+// cilium_sock_term_filter map. mapPinPath specifies the pin path for the
+// cilium_lb4_reverse_sk and cilium_lb4_reverse_sk maps. If mapPinPath is empty,
+// it uses the default pin path.
+func LoadSockTerm(mapPinPath string) (*ebpf.Program, *socktermfilter.Map, error) {
+	spec, err := bpf.LoadCollectionSpec(BPFSockTermPath)
+	if err != nil {
+		return nil, nil, fmt.Errorf("load eBPF ELF: %w", err)
+	}
+
+	// Since we don't compile bpf_sock_term.o at runtime, we need to adjust
+	// MaxEntries for both maps to match the sizes defined in bpf_sock.c.
+	if m := spec.Maps[lbmap.SockRevNat4MapName]; m == nil {
```

Comment:
To be on the safe side, we should always let datapath loading create these maps because they're still suscept to compile-time variance. Right now, if `LRU_MEM_FLAVOR` changes during agent startup, this will still result in an incompatible map and this will potentially recreate and re-pin.

We need to avoid this kind of breakage at all costs; this is an easy trap to fall into and will slip through reviews. The unfortunate soul deploying this into their environment will bear the headache. There's an open issue for a similar mechanism (https://github.com/cilium/cilium/issues/38720), but with that proposal the risk of regression is low since the adjustment logic remains centralized.